src/transporters/amqp.js

```diff
@@ -0,0 +1,329 @@
+/*
+ * moleculer
+ * Copyright (c) 2017 Ice Services (https://github.com/ice-services/moleculer)
+ * MIT Licensed
+ */
+
+"use strict";
+
+const Promise		= require("bluebird");
+const Transporter 	= require("./base");
+const {
+	PACKET_REQUEST,
+	PACKET_RESPONSE,
+	PACKET_UNKNOW,
+	PACKET_EVENT,
+	PACKET_DISCOVER,
+	PACKET_INFO,
+	PACKET_DISCONNECT,
+	PACKET_HEARTBEAT,
+} = require("../packets");
+
+/**
+ * Transporter for AMQP
+ *
+ * More info: https://www.amqp.org/
+ *
+ * @class AmqpTransporter
+ * @extends {Transporter}
+ */
+class AmqpTransporter extends Transporter {
+
+	/**
+	 * Creates an instance of AmqpTransporter.
+	 *
+	 * @param {any} opts
+	 *
+	 * @memberOf AmqpTransporter
+	 */
+	constructor(opts) {
+		if (typeof opts === "string")
+			opts = { amqp: { url: opts } };
+
+		// Number of requests a broker will handle concurrently
+		if (typeof opts.amqp.prefetch !== "number")
```

Comment:
@WoLfulus If opts was a string it should have already been reassigned to an object by this point. The first if-statement in the constructor should handle this case, I think.